fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -463,15 +481,19 @@ protected <M extends Message> CompletableFuture<FDBStoredRecord<M>> saveTypedRec
                             LogMessageKeys.EXPECTED_TYPE, recordType.getName());
                 }
             }
-            final FDBStoredRecord<M> newRecord = serializeAndSaveRecord(typedSerializer, recordBuilder, metaData, oldRecord);
-            if (oldRecord == null) {
-                addRecordCount(metaData, newRecord, LITTLE_ENDIAN_INT64_ONE);
+            if (isDryRun) {
+                return CompletableFuture.completedFuture(recordBuilder.build());
```

Comment:
The record being built here will not have its (fake) new size set correctly. This is probably needed as the customer is interested in knowing the byte-size of the records.

Reply:
good point, I changed the `serializeAndSaveRecord` method to include a dryRun option.

Reply:
I guess I'm less sure that we care about having the exact right `SizeInfo` values here. If we can get the right values, that's great (and may make comparisons with alternative update approaches cleaner), but if it ends up being too difficult to get right, it doesn't seem like the worst thing to jettison.